tekton_configs/pytest-task.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: pytest
+  namespace: gabrielwen
+  annotations:
+    sidecar.istio.io/inject: "false"
+spec:
+  inputs:
+    params:
+    - name: test-worker-img
+      type: string
+      description: Test worker image.
+      default: gcr.io/kubeflow-ci/test-worker:latest
+    - name: src-dir
+      type: string
+      description: The path for source codes.
+      default: /mnt/testdir/src
+    - name: repos
+      type: string
+      description:
+        Repositories we need to checkout.
+    - name: python-path
+      type: string
+      description:
+        PYTHONPATH env.
+    - name: gcp-sa-dir
+      type: string
+      description:
+        GOOGLE_APPLICATION_CREDENTIALS path.
+      default: /secrets/sa
+    - name: get-cred-cmd
+      type: string
+      description:
+        Command used to run get-credentials.
+      default: python3
+    - name: get-cred-args-array
+      type: array
+      description:
+        Args passed into get-credentials cmd.
+    - name: pytest-args-array
+      type: array
+      description:
+        Args passed into pytest.
+    - name: pytest-wdr
+      type: string
+      description:
+        WorkingDir of pytest.
+    - name: test_job_type
+      type: string
+      description:
+        Job type of the test, from prow.
+      default: manual
+    - name: test_hostname
+      type: string
+      description:
+        host name of the test, from prow.
+      default: manual-testing
+  steps:
+  - name: git-checkout
+    image: $(inputs.params.test-worker-img)
+    command: ["/usr/local/bin/checkout_repos.sh"]
+    args:
+    - --repos=$(inputs.params.repos)
+    - --src_dir=$(inputs.params.src-dir)
+    volumeMounts:
+    - mountPath: $(inputs.params.src-dir)
+      name: testing-dir
+  - name: get-credentials
+    image: $(inputs.params.test-worker-img)
+    command: ["$(inputs.params.get-cred-cmd)"]
```

Comment:
Why is the command a parameter? Why can't we just hard code it?

Reply:
since this is a generic pytest task, I'm thinking we might need different commands - for notebook test we use get_kf_cluster but for tests in kfctl we need to get the credential from the testing cluster.